baseapp/baseapp.go

```diff
@@ -732,38 +739,91 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte) (gInfo sdk.GasInfo, re
 	// and we're in DeliverTx. Note, runMsgs will never return a reference to a
 	// Result if any single message fails or does not have a registered Handler.
 	result, err = app.runMsgs(runMsgCtx, msgs, mode)
-	if err == nil {
-		// Run optional postHandlers.
-		//
-		// Note: If the postHandler fails, we also revert the runMsgs state.
-		if app.postHandler != nil {
-			// The runMsgCtx context currently contains events emitted by the ante handler.
-			// We clear this to correctly order events without duplicates.
-			// Note that the state is still preserved.
-			postCtx := runMsgCtx.WithEventManager(sdk.NewEventManager())
-
-			newCtx, err := app.postHandler(postCtx, tx, mode == runTxModeSimulate, err == nil)
-			if err != nil {
-				return gInfo, nil, anteEvents, priority, err
-			}
 
-			result.Events = append(result.Events, newCtx.EventManager().ABCIEvents()...)
+	// Case 1: the msg errors and the post handler is not set.
+	if err != nil && app.postHandler == nil {
+		return gInfo, result, anteEvents, priority, err
+	}
+
+	// Case 2: tx errors and the post handler is set. Run PostHandler and revert state from runMsgs
+	if err != nil && app.postHandler != nil {
```

Comment:
I would combine the `err != nil` into a single clause as it makes it easier to read.

Reply:
@alexanderbez done can you please re-review?